### ISSUE TITLE
[QUESTION]: Script exemplifying cross-validation and hyperparameter optimisation approach?

### ISSUE THREAD

**Issue by jepdavidson** [chemprop/chemprop#311]
**What are you trying to do?**
I have only just started using Chemprop, and thought a good place to start would be to reproduce some of the work in [Analyzing Learned Molecular Representations for Property Prediction](https://doi.org/10.1021/acs.jcim.9b00237).

However, I seem to be struggling to fully understand the exact approach used for cross-validation and hyperparameter optimisation...

In the paper, the experimental section states:

>  Specifically, for each dataset, we use 20 iterations of Bayesian optimization on
10 randomly-seeded 80:10:10 data splits to determine the best hyperparameters, selecting
hyperparameters based on validation set performance. We then evaluate the model by retraining using the optimal hyperparameters and checking performance on the test set.

and the chemprops docs note:

> Note that the hyperparameter optimization script sees all the data given to it. The intended use is to run the hyperparameter optimization script on a dataset with the eventual test set held out. If you need to optimize hyperparameters separately for several different cross validation splits, you should e.g. set up a bash script to run hyperparameter_optimization.py separately on each split's training and validation data with test held out.

To put me on the right track, **I wonder if there is a link somewhere to the original scripts used for the publication** (I couldn't find it in the SI), or if someone is able to share a suitable shell script or, ideally, python api script?

Kind regards

James

**Comment by kevingreenman** [comment]
Hi James, thanks for your question. Chemprop has undergone a lot of development since the original 2019 paper, including some changes to the way hyperparameter optimization works. The authors of the paper have since moved on and are no longer actively involved in maintaining the repo, and I'm unfortunately not aware of anywhere that the original scripts are publicly available.

I'll close this issue for now, but feel free to reopen if you have more questions about how to use the hyperparameter optimization or cross-validation in their current form.
